arwen/elrondapi/elrondei.go

```diff
@@ -2019,12 +2033,17 @@ func StorageLoadFromAddressWithHost(host arwen.VMHost, addressOffset int32, keyO
 }
 
 // StorageLoadFromAddressWithTypedArgs - storageLoadFromAddress with args already read from memory
-func StorageLoadFromAddressWithTypedArgs(host arwen.VMHost, address []byte, key []byte) []byte {
+func StorageLoadFromAddressWithTypedArgs(host arwen.VMHost, address []byte, key []byte) ([]byte, error) {
 	storage := host.Storage()
 	metering := host.Metering()
-	data, usedCache := storage.GetStorageFromAddress(address, key)
-	storage.UseGasForStorageLoad(storageLoadFromAddressName, metering.GasSchedule().ElrondAPICost.StorageLoad, usedCache)
-	return data
+	data, trieDepth, usedCache := storage.GetStorageFromAddress(address, key)
```

Comment:
since we are using almost the same pattern: get storage & trie depth and then compute the required gas, why not merge the 2 instructions in one function?
Bonus, I would attach it to the arwen.VMHost since you always need the EnableEpochsHandler

Reply:
Some functions use `GetStorageUnmetered()`, others use `GetStorage()` and others `GetStorageFromAddress()` Will attach `GetStorageLoadCost()` to storage context.